pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2704,6 +2706,12 @@ func fetchEnvVariablesFromCloudInit(config types.DomainConfig) (map[string]strin
 // Create a isofs with user-data and meta-data and add it to DiskStatus
 func createCloudInitISO(config types.DomainConfig) (*types.DiskStatus, error) {
 
+	if config.CipherInfo != nil && len(config.CipherTextUserData) != 0 {
+		userData, err := cipherDecrypt(config.CipherInfo, config.CipherTextUserData)
+		if len(userData) != 0 && err == nil {
+			config.CloudInitUserData = &userData
```

Comment:
Note that this change to config is local to this function. It would be less confusing if you instead put it in a local variable.